packages/transaction-controller/src/etherscan.ts

```diff
@@ -189,6 +192,11 @@ function getEtherscanApiUrl(
   const apiUrl = `https://${networkInfo.subdomain}.${networkInfo.domain}`;
   let url = `${apiUrl}/api?`;
 
+  // We only support API keys for the Etherscan domain
+  if (networkInfo.domain !== DEFAULT_ETHERSCAN_DOMAIN) {
```

Comment:
I think this could be a logical addition should we need API keys for other domains in future.

Though I fear for the time being, while we're confident only the mobile client wants to use an API key and only on the Etherscan domain, it would add more complexity than benefit.

This is because to actually utilise it, we'd need an additional property in these Etherscan function request types, plus in the `RemoteTransactionSource` type, and `IncomingTransactionHelper`, and `TransactionController` so the client can make the decision about which domain the API key is for.

Plus if the requirement for API keys on different domains did evolve, it might be cleaner just to provide a `Map` of API keys per domain for example so the request can use whatever is suitable and available.